api/src/controllers/map-info-controller.js

```diff
@@ -0,0 +1,220 @@
+// const mongoose = require('mongoose');
+// const ObjectID = require('mongodb').ObjectID;
+
+const defaultLog = require('../utils/logger')('mapLayerInfo');
+const queryActions = require('../utils/query-actions');
+const { ApplicationRoles } = require('../utils/constants/misc');
+const Post = require('../controllers/post/post');
+const Get = require('../controllers/get/get');
+const Delete = require('../controllers/delete/delete');
+const PutUtils = require('../utils/put-utils');
+const { mapLayerInfo: MapLayerInfo } = require('../models/index');
+
+exports.protectedOptions = function(args, res, next) {
+  res.status(200).send();
+};
+
+exports.publicGet = async function(args, res, next) {
+  let mapInfoId = null;
+  let errorMsg = null;
+  if (args.swagger.params.mapInfoId && args.swagger.params.mapInfoId.value) {
+    mapInfoId = args.swagger.params.mapInfoId.value;
+  } else {
+    errorMsg = `publicGet - you must provide an id to get`;
+    defaultLog.info(errorMsg);
+    return queryActions.sendResponse(res, 400, errorMsg);
+  }
+
+  let obj = null;
+  try {
+    obj = await Get.findById(mapInfoId);
+  } catch (error) {
+    errorMsg = `publicGet - error getting map info: ${mapInfoId}`;
+    defaultLog.info(errorMsg);
+    defaultLog.debug(error);
+    return queryActions.sendResponse(res, 400, errorMsg);
+  }
+
+  if (!obj) {
+    errorMsg = `protectedGet - map info not found: ${mapInfoId}`;
+    defaultLog.info(errorMsg);
+    return queryActions.sendResponse(res, 404, errorMsg);
+  }
+
+  return queryActions.sendResponse(res, 200, obj);
+};
+
+exports.protectedGet = async function(args, res, next) {
+  let mapInfoId = null;
+  let errorMsg = null;
+  if (args.swagger.params.mapInfoId && args.swagger.params.mapInfoId.value) {
+    mapInfoId = args.swagger.params.mapInfoId.value;
+  } else {
+    errorMsg = `protectedGet - you must provide an id to get`;
+    defaultLog.info(errorMsg);
+    return queryActions.sendResponse(res, 400, errorMsg);
+  }
+
+  let obj = null;
+  try {
+    obj = await Get.findById(mapInfoId);
+  } catch (error) {
+    errorMsg = `protectedGet - error getting map info: ${mapInfoId}`;
+    defaultLog.info(errorMsg);
+    defaultLog.debug(error);
+    return queryActions.sendResponse(res, 400, errorMsg);
+  }
+
+  if (!obj) {
+    errorMsg = `protectedGet - map info not found: ${mapInfoId}`;
+    defaultLog.info(errorMsg);
+    return queryActions.sendResponse(res, 404, errorMsg);
+  }
+
+  return queryActions.sendResponse(res, 200, obj);
+};
+
+async function getMapLayerInfoSegment(segment) {
+  return await MapLayerInfo.findOne({ _schemaName: 'MapLayerInfo', segment });
+}
+
+exports.protectedPost = async function(args, res, next) {
+  let incomingObj = {};
+  let errorMsg = null;
+  // Only accpet applicaiton: 'LNG' for now
```

Comment:
sp.

Reply:
Fix in https://github.com/bcgov/NRPTI/pull/790/commits/aa0dec560bb5f665fcf3ba1a7c25037c5b122711